src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.Windows.cs

```diff
@@ -199,7 +199,23 @@ public static long CompletedWorkItemCount
         {
             get
             {
-                return ThreadPool.UseWindowsThreadPool ? WindowsThreadPool.CompletedWorkItemCount : PortableThreadPool.ThreadPoolInstance.CompletedWorkItemCount;
+                long count =
+                    UseWindowsThreadPool
+                        ? WindowsThreadPool.CompletedWorkItemCount
+                        : PortableThreadPool.ThreadPoolInstance.CompletedWorkItemCount;
+
+                // Ensure that the returned value is monotonically increasing
+                long lastCount = s_lastCompletedWorkItemCount;
+                if (count > lastCount)
+                {
+                    s_lastCompletedWorkItemCount = count;
```

Comment:
Do we want to use interlocked read/writes to make this reliable on 32-bit platforms?